lib/rubocop/cop/correctors/for_to_each_corrector.rb

```diff
@@ -35,7 +35,15 @@ def collection_source
       end
 
       def requires_parentheses?
-        collection_node.range_type?
+        collection_node.range_type? || send_with_operator? || using_logical_operator?
```

Comment:
Maybe it won't lose information even if it's inline. The name `using_logical_operator?` may be a little confused with `logical_operator?`, but this solves it :-)

```suggestion
        return true if collection_node.send_type? && collection_node.operator_method?
        
        collection_node.range_type? || collection_node.or_type? || collection_node.and_type?
```